src/gluonts/dataset/loader.py

```diff
@@ -67,13 +65,14 @@ def __init__(
         dataset: Dataset,
         *,
         transform: Transformation,
+        cyclic: bool,
         is_train: bool,
         batch_size: int,
         ctx: mx.Context,
-        dtype: DType = np.float32,
-        cyclic: bool = False,
+        dtype: Optional[DType] = np.float32,
```

Comment:
Why is this optional? If I go up the hierarchy of constructions, I end up in the ParallelDataLoader that has this argument optional, and if this is None then sets it to np.float32. Maybe None should not be allowed, and np.float32 be the default in all these classes